spec/support/fake_analytics.rb

```diff
@@ -112,12 +112,49 @@ def browser_attributes
   end
 
   failure_message do |actual|
-    <<~MESSAGE
-      Expected that FakeAnalytics would have received event #{event_name.inspect}
-      with #{attributes.inspect}.
+    matching_events = actual.events[event_name]
+    if matching_events&.length == 1 && attributes.instance_of?(Hash)
+      # We found one matching event. Let's show the user a diff of the actual and expected
+      # attributes
+      expected = attributes
+      actual = matching_events.first
+      message = ''
+      message += "expected: #{expected}\n"
+      message += "     got: #{actual}\n\n"
+      message += "Diff:#{differ.diff(actual, expected)}"
+      message
+    elsif matching_events&.length == 1 && attributes.instance_of?(RSpec::Matchers::BuiltIn::Include)
```

Comment:
Hopefully a side-effect of this feature is to encourage the use of these matchers for partial matchers, in support of [a ticket created a while back](https://cm-jira.usa.gov/browse/LG-5293) to change the default behavior of the hash-based matching.

Reply:
Yeah, and at least it'll be easier to implement that ticket now since the diffs of any existing tests that end up broken will be easier to parse

Reply:
> Yeah, and at least it'll be easier to implement that ticket now since the diffs of any existing tests that end up broken will be easier to parse

Yep, though realistically the only way I see that ticket actually getting implemented is with a mass-conversion of existing hash assertions to equivalent `include(...)`.

Reply:
what if we renamed this matched `have_logged_event_including` and create a new `have_logged_event_exactly`?